html/solve.html

```diff
@@ -1,8 +1,64 @@
+<!-- BootStrap JavaScript imports -->
+<script src="https://code.jquery.com/jquery-3.3.1.slim.min.js" integrity="sha384-q8i/X+965DzO0rT7abK41JStQIAqVgRVzpbzo5smXKp4YfRvH+8abtTE1Pi6jizo" crossorigin="anonymous"></script>
+<script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.14.7/umd/popper.min.js" integrity="sha384-UO2eT0CpHqdSJQ6hJty5KVphtPhzWj9WO1clHTMGa3JDZwrnQq4sF86dIHNDz0W1" crossorigin="anonymous"></script>
+<script src="https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/js/bootstrap.min.js" integrity="sha384-JjSmVgyd0p3pXB1rRibZUAYoIIy6OrQ6VrjIEaFf/nJGzIxFDsf4x0xIM+B07jRM" crossorigin="anonymous"></script>
+
+<!-- CodeMirror and Interpreter JavaScript imports -->
 <script src="lib/codemirror/lib/codemirror.js"></script>
 <script src="lib/codemirror/mode/javascript/javascript.js"></script>
+<script src="lib/js_interpreter/acorn_interpreter.js"></script>
+
+<!-- Local JavaScript import -->
 <script src="scripts/solve.js"></script>
 
+
+<!-- BootStrap CSS imports -->
+<link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/css/bootstrap.min.css" integrity="sha384-ggOyR0iXCbMQv3Xipma34MD+dH/1fQ784/j6cY/iJTQUOhcWr7x9JvoRxT2MZw1T" crossorigin="anonymous">
 <link rel="stylesheet" href="lib/codemirror/lib/codemirror.css">
 
-<div id="code-area"></div>
-<div id="output-area"></div>
+<!-- Local CSS imports -->
+<link rel="stylesheet" href="stylesheets/solve.css">
+
+
+<div class="row fill no-gutters">
+
+  <div class="col-3">
+    <div class="card fill">
+      <div class="card-header">
+        <h5 class="card-title" id="question-title"></h5>
+      </div>
+      <div class="card-body">
+        <p class="card-text" id="question-text"></p>
+      </div>
+    </div>
+  </div>
+
+  <div class="col-6" id="code-area"></div>
+
+  <div class="col-3">
+    <div class="card fill">
+      <div class="card-header">
+
+        <ul class="nav nav-tabs card-header-tabs" id="myTab" role="tablist">
+          <li class="nav-item">
+            <a class="nav-link active" id="output-tab" data-toggle="tab" href="#output" role="tab" aria-controls="output" aria-selected="true">Output</a>
+          </li>
+          <li class="nav-item">
+            <a class="nav-link" id="analysis-tab" data-toggle="tab" href="#analysis" role="tab" aria-controls="analysis" aria-selected="false">Analysis</a>
+          </li>
+        </ul>
+      </div>
+      <div class="card-body">
+        <div class="tab-content" id="myTabContent">
+          <div class="tab-pane fade show active" role="tabpanel" aria-labelledby="output-tab">
+            <pre id="code-output"></pre>
```

Comment:
quick q, why did you use the `pre` tag instead of `p?`

Reply:
I changed it to `pre` so we can keep the formatting of analysis and code output better.
I tried `p`  earlier and it messed up newlines, tabs, spaces, etc.